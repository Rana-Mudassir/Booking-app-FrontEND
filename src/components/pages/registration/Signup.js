import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { signupAsync } from '../../../store/authentication/signup';

const Signup = () => {
  const dispatch = useDispatch();
  const {
    loading, error, user, token,
  } = useSelector((state) => state.signup);

  const [userData, setUserData] = useState({
    name: '', email: '', password: '', password_confirmation: '',
  });

  const [signupError, setSignupError] = useState(null); // State for signup error

  const handleSignup = () => {
    if (!userData.name || !userData.email || !userData.password || !userData.password_confirmation) {
      // Set an error message if any of the fields are empty
      setSignupError("Please fill in all fields.");
      return;
    }

    dispatch(signupAsync(userData));
    // Clear any previous signup errors
    setSignupError(null);
  };

  return (
    <div className="login">
      <div className="login_sub">
        <h2 className="head2">Signup</h2>
        {loading && <p>Loading...</p>}
        {error && (
          <p>
            Error:
            {error}
          </p>
        )}
        {user && token && (
          <p>
            Signup successful. Welcome,
            {user.name}
            !
          </p>
        )}
        {signupError && (
          <p style={{ color: 'red' }}>{signupError}</p> // Display signup error
        )}
        <input
          className="login_input"
          type="text"
          placeholder="Username"
          value={userData.name}
          onChange={(e) => setUserData({ ...userData, name: e.target.value })}
        />
        <br />
        <br />
        <input
          className="login_input"
          type="email"
          placeholder="Email"
          value={userData.email}
          onChange={(e) => setUserData({ ...userData, email: e.target.value })}
        />
        <br />
        <br />
        <input
          className="login_input"
          type="password"
          placeholder="Password"
          value={userData.password}
          onChange={(e) => setUserData({ ...userData, password: e.target.value })}
        />
        <br />
        <br />
        <input
          className="login_input"
          type="password"
          placeholder="Password Confirmation"
          value={userData.password_confirmation}
          onChange={(e) => setUserData({ ...userData, password_confirmation: e.target.value })}
        />
        <br />
        <br />
        <button className="loginbtn" type="button" onClick={handleSignup}>Signup</button>
      </div>
    </div>
  );
};

export default Signup;
