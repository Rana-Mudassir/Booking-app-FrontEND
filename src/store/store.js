import { configureStore } from '@reduxjs/toolkit';
import loginReducer from './authentication/login';
import signupReducer from './authentication/signup';
import aeroplaneReucer from './aeroplane/aeroplane';
import aeroplaneslistReducer from './aeroplaneList/aeroplaneListSlice';
import selectedAirplaneSliceReducer from './aeroplaneList/selectedAirplaneSlice';

const store = configureStore({
  reducer: {

    login: loginReducer,
    signup: signupReducer,
    aeroplane: aeroplaneReucer,
    aeroplaneslist: aeroplaneslistReducer,
    selectedAirplane: selectedAirplaneSliceReducer,
  },
});

export default store;
