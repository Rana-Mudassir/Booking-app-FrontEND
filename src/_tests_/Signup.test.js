import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../store/store';
import Signup from '../components/pages/registration/Login';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Signup />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
