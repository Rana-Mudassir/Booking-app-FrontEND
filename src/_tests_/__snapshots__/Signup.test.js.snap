// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="login"
>
  <div
    className="login_sub"
  >
    <h2
      className="head2"
    >
      Login
    </h2>
    <div>
      <input
        className="login_input"
        onChange={[Function]}
        placeholder="Username"
        type="text"
        value=""
      />
      <br />
      <br />
      <input
        className="login_input"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <br />
      <br />
      <button
        className="loginbtn"
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
